src/Octoshift/CodeScanningAlertService.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Octoshift.Models;
+using OctoshiftCLI;
+
+namespace Octoshift
+{
+    public class CodeScanningAlertService
+    {
+        private readonly GithubApi _sourceGithubApi;
+        private readonly GithubApi _targetGithubApi;
+        private readonly OctoLogger _log;
+
+        public CodeScanningAlertService(GithubApi sourceGithubApi, GithubApi targetGithubApi, OctoLogger octoLogger)
+        {
+            _sourceGithubApi = sourceGithubApi;
+            _targetGithubApi = targetGithubApi;
+            _log = octoLogger;
+        }
+
+        public virtual async Task MigrateCodeScanningAlerts(string sourceOrg, string sourceRepo, string targetOrg,
+            string targetRepo, bool dryRun)
+        {
+            var defaultBranch = await _sourceGithubApi.GetDefaultBranch(sourceOrg, sourceRepo);
+            _log.LogInformation($"Found default branch: {defaultBranch} - migrating code scanning alerts only of this branch.");
+            await MigrateAnalyses(sourceOrg, sourceRepo, targetOrg, targetRepo, defaultBranch, dryRun);
+            await MigrateAlerts(sourceOrg, sourceRepo, targetOrg, targetRepo, defaultBranch, dryRun);
+        }
+
+        protected internal virtual async Task MigrateAnalyses(string sourceOrg, string sourceRepo, string targetOrg,
+            string targetRepo, string branch, bool dryRun)
+        {
+            _log.LogInformation($"Migrating Code Scanning Analyses from '{sourceOrg}/{sourceRepo}' to '{targetOrg}/{targetRepo}'");
```

Comment:
I wonder if we should make the logging more detailed here so it's clear what GitHub instance we're migrating from and to - it might just make these logs friendlier to read.

Reply:
The Problem here is that I don't have that information in that particular context / class, as the class itself only gets a finished `sourceGithubApi` and `targetGithubApi` object (which hides the server information)

To be able to log the instance here, I guess I would either have to:

- Pass the value as an argument just for logging, which feels weird
- Make it accessible from the `GitHubApi`-class - which also feels a bit weird as I guess this is a `private` field there for a reason

Happy to do either, but I guess this is something that I wanted @dylan-smith to look at first and tell me if we can/should do this and how. 🙂